app/models/content_page.rb

```diff
@@ -16,7 +16,53 @@ class ContentPage < ApplicationRecord
 
   before_save :set_slug_from_title
 
+  after_commit do
+    ContentPage.reorder
```

Comment:
What is the purpose of reordering on an `after_commit` callback? It doesn't seem like ordering is persisted.

Reply:
Without reordering, the navigation ignores new pages

The ordering is persisted, in the class variable

Reply:
Right, understood. It feels like this behaviour could be moved out into a `ContentPageOrderService` or something. Isolating the ordering logic away from the model.
Then the model can tell the service to break the cache and reorder or ask the service to fetch who my next and previous pages are.

Reply:
Honestly, I think acts_as_nested_set is the way, then each page gets a next and previous link.  But for now, it does what it needs to, and I hope we can refactor it later

Reply:
Honestly, I think acts_as_nested_set is the way, then each page gets a next and previous link.  But for now, it does what it needs to, and I hope we can refactor it later

Reply:
From what I understand of `acts_as_nested_set` - it requires the model to have 2 fields in the database that are populated. So when a page ordering is changed,  it'll require updates to models the change effects, cascading down the tree. 

I guess the benefit is an order doesn't need to be stored in memory.

Reply:
I assumed that acts_as_nested_set would recalculate its idea of next and previous whenever the position changed.  I only skimmed the docs, but thats what I'd guess.